### PR TITLE
#176: ttrt api changes and feature additions 

### PR DESCRIPTION
**Pull request by tapspatel**

### CODE REVIEW

File: test/ttmlir/Silicon/multiple_add_with_loc.mlir
Comment (nsmithtt) [inline test/ttmlir/Silicon/multiple_add_with_loc.mlir:1]: Can we introduce an additional directory level that corresponds to the intended backend target `test/ttmlir/Silicon/TTNN`?

We'll probably want to organize further once the test list grow, but I'm not sure what that looks like yet.

File: test/ttmlir/Silicon/multiple_add_with_loc.mlir [hunk 1]
```diff
@@ -0,0 +1,27 @@
+// RUN: ttmlir-opt --ttir-to-ttnn-backend-pipeline %s | ttmlir-translate --ttnn-to-flatbuffer > %s.ttnn
```
Comment (nsmithtt) [inline test/ttmlir/Silicon/multiple_add_with_loc.mlir:1]: I think that `%t.ttnn` might be what we want right?  Just as a rule of thumb we don't want build or test artifacts to show up in the source tree.

**Comment by nsmithtt** [comment]
This looks awesome! Couple of minor comments inline, also is there an example of how to run a directory of flatbuffers?

**Comment by tapspatel** [comment]
made all changes and added example of how to run flatbuffer directories

**Review by nsmithtt — APPROVED** [review]

**Pull request merged**